src/NLopt.jl

```diff
@@ -216,7 +216,7 @@ function chk(o::Opt, result::Result)
             e = nlopt_exception
             if e !== nothing && !isa(e, ForcedStop)
                 nlopt_exception = nothing
-                rethrow(e)
+                throw(e)
```

Comment:
Too bad we lose the original stack trace; I guess in the future we could also potentially save the `catch_backtrace()` and wrap everything in our own exception type that shows this along with the original exception.

Reply:
I think wrapping the exception type might break try-catch blocks around the optimize call that expect specific exception types to be thrown.

I tried this: https://github.com/ikirill/NLopt.jl/commit/c7aed006b0cb5433de41f650277a451453b00e42